salt/states/pkg.py

```diff
@@ -1690,45 +1690,45 @@ def installed(
             # checks reinstall targets works.
             pkg_ret = {}
 
-        if 'pkg.hold' in __salt__:
-            if 'hold' in kwargs:
-                try:
-                    if kwargs['hold']:
-                        hold_ret = __salt__['pkg.hold'](
-                            name=name, pkgs=pkgs, sources=sources
-                        )
-                    else:
-                        hold_ret = __salt__['pkg.unhold'](
-                            name=name, pkgs=pkgs, sources=sources
-                        )
-                except (CommandExecutionError, SaltInvocationError) as exc:
-                    comment.append(str(exc))
+    if 'pkg.hold' in __salt__:
+        if 'hold' in kwargs:
```

Comment:
No need in nested ifs and unnecessary large indent because of it:
```python
if 'pkg.hold' in __salt__ and 'hold' in kwargs:
```

Reply:
While @garethgreenaway didn't originally write this (he just dedented existing code), this would probably be a good idea, especially since there is no action taken when the `hold` argument is not passed.